src/main/java/Task.java

```diff
@@ -0,0 +1,48 @@
+import java.util.Date;
+
+/**
+ * Task is part of a TaskList.
+ */
+
+public abstract class Task {
+
+    protected String description;
+    protected boolean isDone;
+    protected final TaskType taskType;
+
+    public Task(String description, TaskType taskType, boolean isDone) {
+        this.description = description;
+        this.taskType = taskType;
+        this.isDone = isDone;
+    }
+
+    public String getStatusIcon() {
+        return "[" + (isDone ? "\u2713" : "\u2718") + "]"; //return tick or X symbols
+    }
+
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    /**
+     * Used by Storage to record the details of a Task.
```

Comment:
I like the explanation on the usage of the method in Javadocs!
However, perhaps adding a short summary of what the method does would be better, to comply with coding standards.

Reply:
Thanks for the comment, I will work on my explanation in Javadocs